src/oscar/apps/order/abstract_models.py

```diff
@@ -223,24 +223,23 @@ def shipping_tax(self):
 
     @property
     def shipping_status(self):
-        events = self.shipping_events.all()
+        events = self.shipping_events.order_by('-date_created').all()
```

Comment:
Although in practice they often are, the common requirement for PKs is only uniqueness. An order on them may not even exist.